vector/src/main/java/fr/gouv/tchap/fragments/PublicRoomsFragment.java

```diff
@@ -0,0 +1,636 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+import butterknife.BindView;
+import im.vector.PublicRoomsManager;
+import im.vector.R;
+import im.vector.activity.CommonActivityUtils;
+import im.vector.activity.RoomDirectoryPickerActivity;
+import im.vector.activity.VectorRoomActivity;
+import im.vector.adapters.AdapterSection;
+import im.vector.adapters.RoomAdapter;
+import im.vector.fragments.AbsHomeFragment;
+import im.vector.util.RoomDirectoryData;
+import im.vector.view.EmptyViewItemDecoration;
+import im.vector.view.SectionView;
+import im.vector.view.SimpleDividerItemDecoration;
+import fr.gouv.tchap.adapters.PublicRoomAdapter;
+
+public class PublicRoomsFragment extends AbsHomeFragment {
+    private static final String LOG_TAG = PublicRoomsFragment.class.getSimpleName();
+
+    // activity result codes
+    private static final int DIRECTORY_SOURCE_ACTIVITY_REQUEST_CODE = 314;
+
+    //
+    private static final String SELECTED_ROOM_DIRECTORY = "SELECTED_ROOM_DIRECTORY";
+
+    // dummy spinner to select the public rooms
+    private Spinner mPublicRoomsSelector;
+
+    private boolean mMorePublicRooms = false;
+    @BindView(R.id.recyclerview)
+    RecyclerView mRecycler;
+
+    // rooms management
+    private PublicRoomAdapter mAdapter;
+
+    // the selected room directory
+    private RoomDirectoryData mSelectedRoomDirectory;
+
+    private  List<String> mCurrentHosts = null;
+    private List<PublicRoomsManager> mPublicRoomsManagers = null;
+    // rooms list
+    private final List<Room> mRooms = new ArrayList<>();
+
+    /*
+     * *********************************************************************************************
+     * Static methods
+     * *********************************************************************************************
+     */
+
+    public static PublicRoomsFragment newInstance() {
+        return new PublicRoomsFragment();
+    }
+
+    /*
+     * *********************************************************************************************
+     * Fragment lifecycle
+     * *********************************************************************************************
+     */
+
+    @Override
+    public View onCreateView(final LayoutInflater inflater, final ViewGroup container, final Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_rooms, container, false);
+    }
+
+    @Override
+    public void onActivityCreated(final Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        mPrimaryColor = ContextCompat.getColor(getActivity(), R.color.tab_rooms);
+        mSecondaryColor = ContextCompat.getColor(getActivity(), R.color.tab_rooms_secondary);
+        String userHSName = mSession.getMyUserId().substring(mSession.getMyUserId().indexOf(":") + 1);
+        List<String> servers = Arrays.asList(getResources().getStringArray(R.array.identity_server_names));
```

Comment:
Why don't you use room_directory_servers array here?

Reply:
no good reason

Reply:
ok I will use room_directory_servers again